endpoint.go

```diff
@@ -22,9 +23,10 @@ func NewEndpoint(s string) *Endpoint {
 		endpoint.Host = parts[1]
 	}
 
-	if parts := strings.Split(endpoint.Host, ":"); len(parts) > 1 {
-		endpoint.Host = parts[0]
-		endpoint.Port, _ = strconv.Atoi(parts[1])
+	host, port, err := net.SplitHostPort(endpoint.Host)
```

Comment:
I don't like how the error is silenced if the input is mangled. It's a breaking change, but I'd prefer that this function propagates the error.

Reply:
It would be nice to use this opportunity to add some unit tests for this as well.

Reply:
Cool, I'm on it

Reply:
I've made the changes, and you were right, in order to propagate error both the signatures for `NewEndpoint` and  `NewSSHTunnel` have to be updated which is a breaking change.